src/edu/washington/escience/myria/parallel/Server.java

```diff
@@ -1057,7 +1066,51 @@ public DatasetStatus deleteDataset(final RelationKey relationKey) throws DbExcep
     }
 
     return getDatasetStatus(relationKey);
+  }
+
+  /**
+   * @param relationKey the relationKey of the dataset to persist
+   * @return the queryID
+   * @throws DbException if there is an error
+   * @throws InterruptedException interrupted
+   */
+  public long persistDataset(final RelationKey relationKey) throws DbException, InterruptedException {
+    long queryID;
+
+    /* Mark the relation as is_persistent */
+    try {
+      catalog.markRelationPersistent(relationKey);
+    } catch (CatalogException e) {
+      throw new DbException(e);
+    }
 
+    /* Create the query plan for persist */
+    try {
+      ImmutableMap<Integer, SubQueryPlan> workerPlans = null;
```

Comment:
I don't understand how this can work--you're overwriting `workerPlans` on each iteration of the `for` loop? You presumably want to declare a new `Builder` here and call `Builder.put()` on each iteration, then call `Builder.build()` after the loop is finished, passing the resulting `ImmutableMap` to `submitQuery()`.